runtime/src/arg.c

```diff
@@ -291,9 +291,12 @@ void parseNumLocales(const char* numPtr, int32_t lineno, int32_t filename) {
       if (_argNumLocalesPerNode < 1) {
         chpl_error("Number of locales per node must be > 0.",
                    lineno, filename);
-      }
-      if (_argNumLocalesPerNode > 1) {
-        chpl_env_set("CHPL_RT_LOCALES_PER_NODE", lpn, 1);
+      } else if ((_argNumLocalesPerNode > 1) || t) {
+        // We have co-locales if there is more then one locale per node or if
+        // the locale is bound to an architectural feature.
+        char buf[32];
+        snprintf(buf, sizeof(buf), "%d", _argNumLocalesPerNode);
+        chpl_env_set("CHPL_RT_LOCALES_PER_NODE", buf, 1);
```

Comment:
Is it possible to reuse some of the logic in `chpl_env_set_uint` here?

```
chpl_env_set_uint("CHPL_RT_LOCALES_PER_NODE", (uint64_t)_argNumLocalesPerNode, 1);
```

Reply:
Fixed, I didn't know about that function.